apps/medon-fe/src/components/PatientCardCalendar/index.tsx

```diff
@@ -0,0 +1,124 @@
+import { useState } from 'react';
+import dayjs from 'dayjs';
+import { useTranslation } from 'react-i18next';
+import { Views, dayjsLocalizer, Event } from 'react-big-calendar';
+import { useTheme } from 'styled-components';
+
+import Link from 'components/Link';
+
+import { eventsCard } from 'utils/mock/patientCalendar';
+import { getDateAndHourEvent } from 'utils/functions/getDateAndHourEvent';
+import { getDayPropGetter } from 'utils/functions/getDayPropGetter';
+import { getEventPropGetter } from 'utils/functions/getEventPropGetter';
+import { roles } from 'utils/constants';
+
+import { useAppSelector } from 'redux/hooks';
+
+import Plus from 'assets/svgs/plus_listcard.svg';
+
+import {
+  Dot,
+  Legend,
+  StyledCalendar,
+  StyledModal,
+  Subtitle,
+  TextSubtitle,
+  Title,
+} from './styles';
+import { useModal } from './hooks';
+import 'react-big-calendar/lib/css/react-big-calendar.css';
+
+export function PatientCardCalendar() {
+  const [event, setEvent] = useState<Event>();
+
+  const theme = useTheme();
+  const { t } = useTranslation();
+  const { role } = useAppSelector((state) => state.userState.user);
+
+  const { hideModal, isVisible, showModal } = useModal(false);
+
+  const localizer = dayjsLocalizer(dayjs);
+  const dayPropGetter = getDayPropGetter(theme);
```

Comment:
you will recalculate it with every rerender, find other way, same for line 40 and 42